### PR TITLE
table: add some injections for testing data-inconsistency defense

### PR DESCRIPTION
**Pull request by ekexium**
Signed-off-by: ekexium <ekexium@gmail.com>

### What problem does this PR solve?

Issue Number: ref #26833

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: table/tables/mutation_checker.go [hunk 1]
```diff
@@ -381,3 +385,87 @@ func getOrBuildColumnMaps(
...
+			}
+		case "corruptIndexKey":
+			// a corrupted index mutation.
+			// TODO: distinguish which part is corrupted, value or handle
+			{
+				indexMutation := indexMutations[0]
+				key := indexMutation.key
+				memBuffer.RemoveFromBuffer(key)
+				key[len(key)-1] += 1
+				if err := memBuffer.Set(key, indexMutation.value); err != nil {
```
Comment (MyonKeminta) [inline table/tables/mutation_checker.go:439]: Should we distinguish putting and deleting like how `extraIndex` does?

**Review by MyonKeminta — APPROVED** [review]

File: table/tables/mutation_checker.go (hunk 1 above)
Comment (cfzjywxk) [reply table/tables/mutation_checker.go:439]: +1

**Review by cfzjywxk — APPROVED** [review]

File: table/tables/mutation_checker.go [hunk 2]
```diff
@@ -381,3 +385,93 @@ func getOrBuildColumnMaps(
...
+			{
+				if len(indexMutations) == 0 {
+					continue
+				}
+				indexMutation := indexMutations[0]
+				value := indexMutation.value
+				if len(value) > 0 {
+					value[len(value)-1] += 1
+				}
+				if err := memBuffer.Set(indexMutation.key, value); err != nil {
```
Comment (MyonKeminta) [inline table/tables/mutation_checker.go:461]: It seems that it's also needed to check if it's a Delete mutation here.

**Review by MyonKeminta — APPROVED** [review]

**Pull request merged**